.github/workflows/main.yml

```diff
@@ -0,0 +1,185 @@
+    - name: Build
+      working-directory: ${{github.workspace}}/build
+      run: NINJA_STATUS="%p [%f:%s/%t] %o/s, %es" ninja
+      # If the tests are enabled for this job, run them
+    - name: Test
+      if: ${{ matrix.build-only != 'yes' }}
+      working-directory: ${{github.workspace}}/build
+      run: LD_LIBRARY_PATH=/usr/${{ matrix.triple }}/lib ctest --output-on-failure -j 4
+
+  windows:
+    strategy:
+      matrix:
+        # Build each combination of OS and release/debug variants
+        os: [ windows-2016, windows-2019 ]
+        build-type: [ Release, Debug ]
+        arch: [ Win32, x64 ]
+        toolchain: [ "", "-T ClangCL" ]
+        extra-cmake-flags: [ "" ]
+        # The ClangCL toolchain was added in Visual Studio 2019, the Windows
+        # 2016 runners have only VS 2017, so skip them for this configuration
+        exclude:
+          - os: windows-2016
+            toolchain: "-T ClangCL"
+        # Add an extra check for the Windows 8 compatible PAL
+        include:
+          - os: windows-2016
+            build-type: Release
+            arch: x64
+            toolchain: ""
+            extra-cmake-flags: -DWIN8COMPAT=TRUE
+            variant: Windows 8 compatible
+      # Don't abort runners if a single one fails
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }} ${{ matrix.arch }} ${{ matrix.build-type }} ${{ matrix.toolchain }} ${{ matrix.variant }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Configure CMake
+      run: |
+        cmake -B ${{github.workspace}}/build -A ${{ matrix.arch }} ${{ matrix.toolchain }} ${{ matrix.extra-cmake-flags }} -DSNMALLOC_CI_BUILD=On -DSNMALLOC_RUST_SUPPORT=On
+      # Build with whatever version of msbuild was configured, for a single build configuration.
+    - name: Build
+      run: cmake --build ${{github.workspace}}/build -- /m /p:Configuration=${{ matrix.build-type }}
+      # Run the tests.
+    - name: Test
+      working-directory: ${{ github.workspace }}/build
+      run: ctest -j 2 --interactive-debug-mode 0 --output-on-failure -C ${{ matrix.build-type }}
+
+
+  # Job to run clang-format and report errors
+  format:
+    runs-on: ubuntu-18.04
+    # We don't need to do the build for this job, but we need to configure it to get the clang-format target
+    steps:
+    - uses: actions/checkout@v2
+    - name: Configure CMake
+      run: cmake -B ${{github.workspace}}/build
+    # Run the clang-format check and error if it generates a diff
+    - name: Run clang-format
+      working-directory: ${{github.workspace}}/build
+      run: |
+        set -eo pipefail
+        make clangformat
+        git diff --exit-code
+    - name: Run clang-tidy
+      run: |
+        #clang-tidy-9  src/override/malloc.cc  -header-filter="`pwd`/*" -warnings-as-errors='*' -export-fixes=tidy.fail -- -std=c++17 -mcx16 -DSNMALLOC_PLATFORM_HAS_GETENTROPY=0
```

Comment:
Is this commented out?

Reply:
clang-tidy isn't installed on the runner.  I'm not sure if it's actually necessary because I couldn't see how the old one was reporting errors (which is slightly embarrassing, because I'm pretty sure I wrote this bit of the old CI system).  Do we ever get errors from this in PRs?  If not, I'll disable it entirely.  If so, I'll try to make it work.

Reply:
I have had it regularly complain at me.  Normally for the better.

My understanding was it generated a fixes file:
```
 -export-fixes=tidy.fail 
```

Then  checked if that existed :
```
        if [ -f tidy.fail ] ; then
          cat tidy.fail
          exit 1
        fi
```

It worked pretty well on ADO.